TypeCobol/Compiler/Preprocessor/PreprocessorStep.cs

```diff
@@ -633,7 +633,7 @@ private static IList<DocumentChange<IProcessedTokensLine>> ProcessTokensLinesCha
                         tokensLineWithCopyDirective.ImportedDocuments.Keys.Where(
                             c => c.TextName != null || c.COPYToken.TokenType == TokenType.EXEC).ToArray())
                     {
-                        if (copyDirective.TextName == null)
+                        if (copyDirective.TextName == null || tokensLineWithCopyDirective.ScanStateBeforeCOPYToken == null)
```

Comment:
Do we have cases where the scan state dictionary is `null` ? The token line with COPY directive should have this dictionary set according to this code:

https://github.com/TypeCobolTeam/TypeCobol/blob/4cc51f941a01dbb62eb94ba0f2db5330d22ffc00/TypeCobol/Compiler/Scanner/TokensLine.cs#L73-L80

So maybe turn it into a `Debug.Assert` ?

Reply:
I really don't know but I put this because it happened. Note that the code that you report is in the AddToken() method, thus if we have for any reason a TokenLine instance in which no token has been added ScanStateBeforeCOPYToken can be null, thus it is not an assertion. But I got such situation, for instance when executing some tests I think.

Reply:
I ran all unit tests in EI_Debug with the second part of the condition commented and encountered no error.

Reply:
I got it,maybe I will be able to reproduce it one day.

Reply:
The condition has already been added for the cup method and it's on develop now, but i checked the tests without it and it works so could you:
- update this branch with latest changes from develop
- remove the condition in cup method and turn it into an assert

Reply:
Ok there is a bug on EXEC SQL INCLUDE, the ScanStateBeforeCOPYToken should be non-null. See #2105.
No need to change the test here in this PR.